python/cudf/cudf/tests/test_scalar.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+import pytest
+
+from cudf._libxx.scalar import Scalar
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        0,
+        -1,
+        42,
+        0.0,
+        1.0,
+        np.int8(0),
+        np.int8(1),
+        np.int8(-1),
+        np.int16(1),
+        np.int32(42),
+        np.int32(-42),
+        np.int64(42),
+        np.float32(1),
+        np.float32(-1),
+        np.float64(1),
+        np.float64(-1),
+        np.datetime64(0, "s"),
+        np.datetime64(1, "s"),
+        np.datetime64(-1, "s"),
+        np.datetime64(42, "s"),
+        np.datetime64(42, "ms"),
+        np.datetime64(42, "us"),
+        np.datetime64(42, "ns"),
+        "",
+        "one",
+        "1",
```

Comment:
Suggest adding a numpy string scalar and possible a numpy object here as well:
```
np.str_("asdf"),
np.object_("asdf"),
```

Reply:
Done -- see below